server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -221,9 +234,21 @@ private String innerDetect() throws Exception {
             if (threadName == null) {
                 continue; // thread is not alive yet or died before the first snapshot - ignore it!
             }
-            double percent = (((double) time) / interval.nanos()) * 100;
-            sb.append(String.format(Locale.ROOT, "%n%4.1f%% (%s out of %s) %s usage by thread '%s'%n",
-                percent, TimeValue.timeValueNanos(time), interval, type, threadName));
+
+            if ("mem".equals(type)) {
+                double percent = 0;
+                MemoryUsage memUsage = memoryMXBean.getHeapMemoryUsage();
```

Comment:
I think it would be better if we reported the amount of memory allocated per thread in bytes, and avoid reporting percentages of total memory used. 

The call to getHeapMemoryUsage() will give us info on how much of the absolute overall heap is currently occupied by objects that are either alive or dead. However, the amount of calculated allocated memory per thread is only what the thread allocated for the last 'interval' we slept for, e.g. 500ms. The percentages reported will be inaccurate. 

We could get more accurate in reporting the percentages, by capturing the memory usage before and after the call to `Thread.sleep(interval.millis())`. However, the JVM comes with many Garbage Collector(GC) technologies, some of the more recent ones are concurrent, which means that they perform at least part of the collection process while the application threads are running. 

Essentially, any GC interference would make the reported before/after 'used memory' unstable. For accurate percentages we'd need to rely on no GC interference which isn't possible.